### PR TITLE
[VTK] fix linux build

### LINKED ISSUE #3828: Error: Building package vtk:x64-linux failed with: POST_BUILD_CHECKS_FAILED
Hi, I was testing installing VTK on my machine and got this error. I tried to check the logs, but I found nothing useful. I tried in two different machines but the error was the same on both machines (with different linux distros). 

Here are the output messages: 

`The following packages will be built and installed:
    vtk[core]:x64-linux
Starting package 1/1: vtk:x64-linux
Building package vtk[core]:x64-linux... 
-- Using cached /home/tf/Libraries/vcpkg/downloads/Kitware-VTK-v8.1.0.tar.gz
-- Using source at /home/tf/Libraries/vcpkg/buildtrees/vtk/src/v8.1.0-438bd232dc
-- Applying patch /home/tf/Libraries/vcpkg/ports/vtk/dont-define-ssize_t.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch /home/tf/Libraries/vcpkg/ports/vtk/use-fixed-find-hdf5.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch /home/tf/Libraries/vcpkg/ports/vtk/disable-workaround-findhdf5.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch /home/tf/Libraries/vcpkg/ports/vtk/fix-find-libproj4.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch /home/tf/Libraries/vcpkg/ports/vtk/fix-find-libharu.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch /home/tf/Libraries/vcpkg/ports/vtk/fix-find-mysql.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch /home/tf/Libraries/vcpkg/ports/vtk/fix-find-odbc.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Performing post-build validation
There should be no empty directories in /home/tf/Libraries/vcpkg/packages/vtk_x64-linux
The following empty directories were found: 

    /home/tf/Libraries/vcpkg/packages/vtk_x64-linux/tools/vtk

If a directory should be populated but is not, this might indicate an error in the portfile.
If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:

    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/a/dir ${CURRENT_PACKAGES_DIR}/some/other/dir)

Found 1 error(s). Please correct the portfile:
    /home/tf/Libraries/vcpkg/ports/vtk/portfile.cmake
-- Performing post-build validation done
Error: Building package vtk:x64-linux failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: vtk:x64-linux
  Vcpkg version: 0.0.113-unknownhash

Additionally, attach any relevant sections from the log files above.`

Is there a fix for this problem?

### LINKED ISSUE #4103: [VTK-8.1] Building package vtk:x64-linux failed with: POST_BUILD_CHECKS_FAILED
When trying to build VTK 8.1 (Linux) at the final stage I get the following error:

    -- Configuring x64-linux-dbg
    -- Configuring x64-linux-rel
    -- Building x64-linux-dbg
    -- Building x64-linux-rel
    -- Performing post-build validation
    There should be no empty directories in /home/xubuntu/Software/vcpkg/packages/vtk_x64-linux
    The following empty directories were found: 

        /home/xubuntu/Software/vcpkg/packages/vtk_x64-linux/tools/vtk

    If a directory should be populated but is not, this might indicate an error in the portfile.
    If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:

        file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/a/dir ${CURRENT_PACKAGES_DIR}/some/other/dir)

    Found 1 error(s). Please correct the portfile:
        /home/xubuntu/Software/vcpkg/ports/vtk/portfile.cmake
    -- Performing post-build validation done
    Error: Building package vtk:x64-linux failed with: POST_BUILD_CHECKS_FAILED
    Please ensure you're using the latest portfiles with `.\vcpkg update`, then submit an issue at https://github.com/Microsoft/vcpkg/issues including:
      Package: vtk:x64-linux
      Vcpkg version: 0.0.113-unknownhash

Although all the files are located in `vtk/lib` folder, `vtk` is not listed as installed in `vcpkg`

### PR DESCRIPTION
**Pull request by kypp**
The Linux build failed due to an empty directory which was empty due to tools not being copied, which weren't copied due to executable extension always specified as `.exe`

fixes #4103
fixes #3828

### CODE REVIEW

File: ports/vtk/portfile.cmake [hunk 1]
```diff
@@ -302,12 +302,17 @@ endforeach()
 # Move executable to tools directory and clean-up other directories
 file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools/vtk)
 
+if (WIN32)
+    set(VTK_EXECUTABLE_EXTENSION .exe)
```
Comment (albertziegenhagel) [inline ports/vtk/portfile.cmake]: I think you can just use [CMAKE_EXECUTABLE_SUFFIX](https://cmake.org/cmake/help/latest/variable/CMAKE_EXECUTABLE_SUFFIX.html) for that.

Comment (kypp) [reply ports/vtk/portfile.cmake]: Thanks, didn't know there was a built-in for that!

I've updated the PR with your suggestion.

**Comment by kypp** [comment]
I have updated this PR to also set the proper tools location in the outputted CMake Targets files.

Without this, using VTK as a dependency on Linux would complain about nonexistent targets (such was the case with `vtk-dicom`).

**Comment by albertziegenhagel** [comment]
I think the problem with the tools directory is related to a fix proposed in MR #3425, where @fbudin69500 simply sets `VTK_INSTALL_RUNTIME_DIR`  to `tools`. This will make sure the tools will be installed into the correct directory in the first place and all references in the CMake config files should be set correctly as well.

There might still be problems with the debug config file though, since the debug tools are getting removed afterwards.

**Comment by fbudin69500** [comment]
Indeed, that was part of my PR, but sadly it never got merged and I never got feedback to know what to improve to get it merged.

File: ports/vtk/portfile.cmake (hunk 1 above)
Comment (Rastaban) [reply ports/vtk/portfile.cmake]: Unfortunately the CMAKE_EXECUTABLE_SUFFIX is not available in the portfiles.  After an offline discussion with @ras0219-msft I am going to change it to a foreach that does not rely on the architecture.

**Comment by Rastaban** [comment]
When running regression tests I see failures in wxwidgets:x64-osx, pcl:x64/x86-windows, and vtk-dicom:x86/x64-windows.

I will also take a look at #3425, for @fbudin69500

**Comment by Rastaban** [comment]
Our clean Linux machine used in regression testing is still fails to build vtk because it is missing X11_Xt_LIB
```
CMake Error at Rendering/OpenGL2/CMakeLists.txt:195 (message):
  X11_Xt_LIB could not be found.  Required for VTK X lib.
```
I am okay with checking this is even with that failure as it is not a regression as long as you verify it works for you.

**Comment by fbudin69500** [comment]
@Rastaban Is there a way to know the configuration of the test machines? Do you use a docker image that we, user, could download to test our builds on?

**Comment by Rastaban** [comment]
@fbudin69500 Unfortunately we don't currently have a way to do share the configuration.  However we are working towards making the test system more accessible to the community.

**Pull request closed without merging**